interface/procedure_tools/labcorp/gen_hl7_order.inc.php

```diff
@@ -355,7 +354,11 @@ function gen_hl7_order($orderid, &$out, &$reqStr)
     $P[70] = $vitals['height'];
     $P[88] = $vitals['bps'] . '^' . $vitals['bpd'];
     $P[89] = $vitals['waist_circ'];
-    $C[17] = hl7Date(date("Ymd", strtotime($porow['date_collected'])));
+    if (!empty($porow['date_collected'])) {
+        $C[17] = hl7Date(date("Ymd", strtotime($porow['date_collected'])));
+    } else {
+        $C[17] = null;
```

Comment:
Not sure hl7 allows null if this is making it to the segment. Would probably blow up parser. Recommend empty as a place holder but perhaps a quick check of the spec is needed to verify.
Wait, I see! data() may generate the good old 1969 date if not null but I forget if it does or doesn't on an empty string so be sure to check the downloaded hl7 file if provider is set to download or or the file in directory which you fix later on.
Sorry for rambling. I may be over explaining like I'm doing now for this sorry!:)

Reply:
@sjpadgett I can set this to be an empty string.  Previously strtotime would return 0 which would set the date_collected to be the current date which seems like a big error. Will an empty string work or does it need to be an HL7 null value code?

Reply:
Yep empty string. Null flavor is a v3 concept where in v2.x empty string is common. However for LabCorp's as with other receivers they may require a special code which I don't think applies here and I no longer have labcorps spec.
Bottom line, set to empty string.:)

Reply:
Changed it to be empty string, once CI finishes, I'll bring it in if there aren't any problems.